### ISSUE TITLE
Don't store files in `~/.jolt`

### ISSUE THREAD

**Issue by nazar-pc** [a16z/jolt#465]
Jolt isn't so special that it needs to store its files in the user profile directory.

Please consider following platform-specific rules, which is easy to implement with https://docs.rs/dirs

**Comment by 0xAndoroid** [comment]
It's common practice to store binaries inside of ~/.directory

**Comment by nazar-pc** [comment]
Yes, it is unfortunately quite common to dump various things into a home directory, disregarding all standards and conventions, but it is not something to be inspired by.

Responsible developers use proper locations according to [XDG specification](https://specifications.freedesktop.org/basedir-spec/latest/) on Linux and similarly idiomatic locations on other operating systems.

`dirs` crate that I suggested originally is abstracting it away behind straightforward cross-platform API.
You should not store binaries under `~/.jolt`, it should probably be something like `~/.local/share/jolt/bin` and maybe a symlink to key entrypoints under `~/.local/bin` if they are supposed to be called directly.

Configs should be under `~/.config/jolt`, caches under `~/.cache/jolt`, state under `~/.local/state/jolt`, etc. The majority of applications actually follow conventions and you can find plenty of examples at those locations on any Linux installation.

Please respect users and don't pollute their home directory with even more things, there are enough irresponsible developers that think they're too important to follow the standards as is.
